src/nbl/asset/utils/CIESProfile.h

```diff
@@ -0,0 +1,137 @@
+                IES_STORAGE_FORMAT getMaxValue() const { return maxValue; }
+
+                const IES_STORAGE_FORMAT& getIntegralFromGrid() const { return integral; }
+                const IES_STORAGE_FORMAT& getAvgEmmision() const { return avgEmmision; }
+
+                template<class ExecutionPolicy>
+                core::smart_refctd_ptr<asset::ICPUImageView> createIESTexture(ExecutionPolicy&& policy, const size_t& width = CDC_DEFAULT_TEXTURE_WIDTH, const size_t& height = CDC_DEFAULT_TEXTURE_HEIGHT) const;
+                core::smart_refctd_ptr<asset::ICPUImageView> createIESTexture(const size_t& width = CDC_DEFAULT_TEXTURE_WIDTH, const size_t& height = CDC_DEFAULT_TEXTURE_HEIGHT) const;
+                
+                template<class ExecutionPolicy>
+                static bool flattenIESTexture(ExecutionPolicy&& policy, const IES_STORAGE_FORMAT& avgEmmision, core::smart_refctd_ptr<asset::ICPUImageView> inoutIES, const float flatten = 0.0);
+                static bool flattenIESTexture(const IES_STORAGE_FORMAT& avgEmmision, core::smart_refctd_ptr<asset::ICPUImageView> inoutIES, const float flatten = 0.0);
+
+            private:
+                CIESProfile(PhotometricType type, size_t hSize, size_t vSize)
+                    : type(type), version(V_SIZE), hAngles(hSize), vAngles(vSize), data(hSize* vSize) {}
+
+                // TODO for @Hazard, I would move it into separate file, we may use this abstraction somewhere too
+                //! Returns spherical coordinates with physics convention in radians
+                /*
+                    https://en.wikipedia.org/wiki/Spherical_coordinate_system#/media/File:3D_Spherical.svg
+                    Retval.first is "theta" polar angle in range [0, PI] & Retval.second "phi" is azimuthal angle
+                    in range [-PI, PI] range
+
+                    Cartesian coordinates obtained from the spherical coordinates in Nabla
+                    are assumed to have radius equal to 1 and therefore always are
+
+                    x = cos(phi)*sin(theta)
+                    y = sin(phi)*sin(theta)
+                    z = cos(theta)
+                */
+
+                static inline std::pair<float, float> sphericalDirToRadians(const core::vectorSIMDf& dir);
+                
+                //! Octahedral coordinate mapping is following
+                /*
+                    center is Z-
+                    U+ from center is X+
+                    V+ from center is Y+
+
+                    when viewed as a texture, the net folds, and the apex where the seams join is Z+
+                */
+                
+                static inline core::vectorSIMDf octahdronUVToDir(const float& u, const float& v);
+                
+                void addHoriAngle(IES_STORAGE_FORMAT hAngle)
+                {
+                    hAngles.push_back(hAngle);
+                    data.resize(getHoriAngles().size() * vAngles.size());
+                }
```

Comment:
hide this, you probably only need it at a single point in time, great candidate for being a lambda or something